lib/deprecation_toolkit/warning.rb

```diff
@@ -38,7 +38,8 @@ def handle_multipart(str)
     end
 
     def deprecation_triggered?(str)
-      DeprecationToolkit::Configuration.warnings_treated_as_deprecation.any? { |warning| warning =~ str }
+      DeprecationToolkit::Configuration.warnings_treated_as_deprecation.any? { |warning| warning =~ str } &&
+        !DeprecationToolkit::Configuration.allowed_warnings.any? { |warning| warning =~ str }
```

Comment:
```suggestion
      DeprecationToolkit::Configuration.warnings_treated_as_deprecation.any? { |matcher| matcher === str }
```

Is what I'm suggesting

Reply:
I don't follow how it will work? 
There is a simple test case that you can use to show me how it should work. 
```ruby
test_cases = {
  'Some useful warning from your code' => false,
  'Another warning' => false,
  'One more warning' => true
}

def my_implementation(str)
  warning_threated_as_deprecations = [//]
  allowed_warnings = [
    /.*Some useful warning from your code.*/,
    /.*Another warning.*/
  ]

  warning_threated_as_deprecations.any? { |warning| warning =~ str } &&
    !allowed_warnings.any? { |warning| warning =~ str }
end

def proposed_implementation(str)
  warning_threated_as_deprecations = [
    //,
    /^(?!Some useful warning from your code$).*/,
    /^(?!Another warning$).*/
  ]
  warning_threated_as_deprecations.any? { |warning| warning === str }
end

def run_test(implementation, test_cases)
  test_cases.keys.each do |test_case|
    raise StandardError.new("Test failed on #{test_case}") unless send(implementation, test_case) === test_cases[test_case]
    puts 'Test passes!'
  end
end

run_test(:my_implementation, test_cases)
run_test(:proposed_implementation, test_cases)
```

Reply:
`===` allow you to pass more than regexps, e.g. procs:

```ruby
NON_DEPRECATION_WARNINGS = Regexp.union(
  /Some useful warning from your code$/,
  /Another warning/,
)
warning_threated_as_deprecations = [
  Proc.new { |str| !str.match?(NON_DEPRECATION_WARNINGS) }
]
```